newIDE/app/src/SceneEditor/index.js

```diff
@@ -76,7 +78,9 @@ export default class SceneEditor extends Component {
       onCloseLayerRemoveDialog: null,
       layerRemoved: null,
 
-      editedObject: null,
+      // editedObject: null, //deprecated over below
+      editedObjectWithContext: { object: null, global: null },
+      editedObjectNewName: null,
```

Comment:
You have a duplicated state in SceneEditor (editedObjectNewName) and in ObjectEditorDialog (newObjectName). You should need only one - the idea of React is to be able to have the state at only one place (this is called having a "single source of truth") to avoid subtle bugs where you forget to update one of them.

Can you try to:
* either put the editedObjectNewName only in SceneEditor, passing it as a props and calling a onChange method that you pass that (a bit like canRenameObject - but which should then be called onChangeObjectName).
* either you choose to put the state in `ObjectEditorDialog`, in which case your remove it from `SceneEditor`. You still pass `canRenameObject`, put now this method is not doing `this.setState` anymore, it's just returning if the object is valid or not. You'll need to pass the object name as a parameter in `onApply`, so that SceneEditor can get it and proceed to the real renaming when the editor is closed.

Choosing one solution or the other is depend on where you want to put the responsibility of the renaming. The second solution could make more sense as the dialog can own the state containing the object name, and pass it to the parent as an argument of onApply.

Reply:
Done